### PR TITLE
ENH: interpolate: addition to `make_lsq_bspline`

### PR DESCRIPTION
**Pull request by Searchingdays**

### CODE REVIEW

File: scipy/interpolate/_bsplines.py [hunk 1]
```diff
@@ -1509,7 +1509,7 @@ def make_interp_spline(x, y, k=3, t=None, bc_type=None, axis=0,
     return BSpline.construct_fast(t, c, k, axis=axis)
 
 
-def make_lsq_spline(x, y, t, k=3, w=None, axis=0, check_finite=True):
+def make_lsq_spline(x, y, t, k=3, w=None, axis=0, check_finite=True, bc_type:None):
```
Comment (ev-br) [inline scipy/interpolate/_bsplines.py:1512]: this is not a legal python syntax.

File: scipy/interpolate/_bsplines.py [hunk 2]
```diff
@@ -1629,6 +1629,16 @@ def make_lsq_spline(x, y, t, k=3, w=None, axis=0, check_finite=True):
...
     y = np.moveaxis(y, axis, 0)    # now internally interp axis is zero
 
+    if bc_type is not None:
+        bc_type = np.asarray_chkfinite(bc_type)
+
+    deriv1 = bc_type[0]
+    deriv2 = bc_type[-1]
+
+    t[0] = deriv1
+    t[-1] = deriv2
```
Comment (ev-br) [inline scipy/interpolate/_bsplines.py:1639]: This cannot be right.

**Review by ev-br — CHANGES_REQUESTED** [review]

**Comment by lucascolley** [comment]
Hi @Searchingdays , please can you follow the PR template and provide a description of why changes, and why you think they should be included? For API additions like this, it is generally a good idea to open an issue first, to discuss with maintainers whether it is something that would belong in SciPy. More info at https://scipy.github.io/devdocs/dev/hacking.html.

**Comment by ev-br** [comment]
The sparse changes are unrelated and should be in a separate PR. The interpolate changes are not really correct. I believe it'd be easier to start from scratch, so am closing this PR. @Searchingdays please don't hesitate resubmitting a focused PR.

**Comment by lucascolley** [comment]
Please read the dev guide at http://scipy.github.io/devdocs/dev/hacking.html and reach out via https://scipy.org/community/ if you'd like any help!

**Pull request closed without merging**